### DISCUSSION TITLE
How to update the version of an AWS Lambda Layer ARN?

### DISCUSSION THREAD

**Discussion by kayman-mk** [renovatebot/renovate#19026] [Suggest an Idea]
Hi,

I am using Renovate for quite a while now. A really great tool. But I am missing support for AWS Lambda Layers. These layers are versioned so it should be possible to update them with Renovate.

The ARN looks like `arn:aws:lambda:<AWS_REGION>:464622532012:layer:dd-trace-java:<VERSION>`. Together with the RegEx manager I can identify them somewhere in my source code (usually Terraform). But the datasource is missing to fetch the version numbers.

Suggestion: Add a `aws-versioned-arn` datasource to return the available version numbers for the ARN. The datasource can be generic. I guess we will find other versioned ARNs in AWS which should be updateable too.

**Comment by rarkins** [comment]
In this case the `arn:aws:lambda:<AWS_REGION>:464622532012:layer:dd-trace-java` string would be `packageName`?

If a new datasource is added then it would enable it to be updated using regex manager.

You can add a feature request issue for it in this repo.

Ref: https://docs.aws.amazon.com/lambda/latest/dg/configuration-versions.html

**Reply by kayman-mk** [reply]
Ok, let's give it a try with #19081

**Reply by jimleroyer** [reply]
Hey folks. What happened to the PR? We are looking for similar functionality and it seemed it was near completion. 👀

**Reply by kayman-mk** [reply]
Almost done with #28251

**Reply by jimleroyer** [reply]
Thanks @kayman-mk ! I didn't see the other PR , thanks for filling me in on that and for working on this. 🙏

**Reply by kevin-kortum-trustedshops** [reply]
Seems like the PR is stale currently
